File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Skills/SkillException.cs
using System;

namespace Microsoft.Bot.Solutions.Skills
{
    public enum SkillExceptionType
    {
        /// <summary>
        ///  Access Denied when calling external APIs
        /// </summary>
        APIAccessDenied,

        /// <summary>
        /// Other types of exceptions
        /// </summary>
        Other
    }

    public class SkillException : Exception
    {
        public SkillException(SkillExceptionType exceptionType, string message, Exception innerException)
            : base(message, innerException)
        {
            ExceptionType = exceptionType;
        }

        public SkillExceptionType ExceptionType { get; set; }
    }
}

File: lib/typescript/botbuilder-skills/src/skillException.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

export enum SkillExceptionType {
    /**
     * Access Denied when calling external APIs
     */
    APIAccessDenied,
    /**
     * Account Not Activated when calling external APIs
     */
    AccountNotActivated,
    /**
     * Other types of exceptions
     */
    Other
}

export class SkillException extends Error {
    public exceptionType: SkillExceptionType;
    public innerException: Error;

    constructor(exceptionType: SkillExceptionType, message: string, innerException: Error) {
        super(message);
        this.exceptionType = exceptionType;
        this.innerException = innerException;
    }
}
